### PR TITLE
WT-9224 Process the call log for close_session

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp [hunk 1]
```diff
@@ -43,13 +43,15 @@ call_log_manager::call_log_manager(const std::string &call_log_file)
 
     _call_log = json::parse(file);
     api_map_setup();
+    _conn = nullptr;
```
Comment (Siddhartha8899) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:46]: @Sean04 I have assigned conn here, to make sure everything is assigned in one place, i.e. constructor.

Comment (etienneptl) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp:46]: You can use the [constructor initializer list](https://en.cppreference.com/w/cpp/language/constructor) for this.
```suggestion
    call_log_manager::call_log_manager(const std::string &call_log_file): _conn(nullptr)
```

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp [hunk 2]
```diff
@@ -64,21 +66,25 @@ call_log_manager::process_call_log_entry(json call_log_entry)
...
+    {
         const std::string session_id = call_log_entry["session_id"].get<std::string>();
-
         /*
          * Not having a valid connection is a fatal error since no other operations can happen
          * without a connection.
          */
         if (_conn == nullptr)
-            throw std::runtime_error("Could not open session, connection does not exist");
+            throw std::runtime_error("Could not open session (Session ID: " + session_id + ")" +
```
Comment (etienneptl) [inline test/simulator/timestamp/call_log_manager/call_log_manager.cpp:82]: np: I would use lower case for `Session` in the message. If you decide to proceed with this change, it appears at different places.
```suggestion
            throw std::runtime_error("Could not open session (session ID: " + session_id + ")" +
```

**Review by etienneptl — APPROVED** [review]

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp (hunk 1 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Thanks for this suggestion, I like the idea of using an initializer list. Updated in [1ccbacb](https://github.com/wiredtiger/wiredtiger/pull/8118/commits/1ccbacb59d7a880c240f5db27ba5676236bafd5c)

File: test/simulator/timestamp/call_log_manager/call_log_manager.cpp (hunk 2 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/call_log_manager/call_log_manager.cpp]: Updated in [cfb4b64](https://github.com/wiredtiger/wiredtiger/pull/8118/commits/cfb4b644efb9b4449b5114e5e75fb14d8ee94a52)

File: test/simulator/timestamp/src/connection_simulator.cpp [hunk 3]
```diff
@@ -46,6 +49,23 @@ connection_simulator::open_session()
...
 
+bool
+connection_simulator::close_session(session_simulator *session)
+{
+    std::vector<session_simulator *>::iterator position =
+      std::find(_session_list.begin(), _session_list.end(), session);
+
+    /* If session doesn't exist in the session_list return false. */
+    if (position == _session_list.end()) {
+        std::cerr << "Error: Session is not present in the session list" << std::endl;
```
[the remark this answers is not in the document]
Comment (Siddhartha8899) [reply test/simulator/timestamp/src/connection_simulator.cpp:60]: Updated in [cfb4b64](https://github.com/wiredtiger/wiredtiger/pull/8118/commits/cfb4b644efb9b4449b5114e5e75fb14d8ee94a52)

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**